vllm/entrypoints/openai/serving_embedding.py

```diff
@@ -24,15 +26,18 @@ def request_output_to_embedding_response(
     request_id: str,
     created_time: int,
     model_name: str,
+    encoding_format: str
 ) -> EmbeddingResponse:
-    data: List[EmbeddingResponseData] = []
+    data = []
     num_prompt_tokens = 0
     for idx, final_res in enumerate(final_res_batch):
         assert final_res is not None
         prompt_token_ids = final_res.prompt_token_ids
-
+        embedding = final_res.outputs.embedding
+        if encoding_format == "base64":
+            embedding = base64.b64encode(np.array(embedding))
         embedding_data = EmbeddingResponseData(
-            index=idx, embedding=final_res.outputs.embedding)
+            index=idx, embedding=[embedding])
```

Comment:
I think here you should either send `List[float]` (float output) or a `str` (base64 output)

Reply:
> I think here you should either send `List[float]` (float output) or a `str` (base64 output)

thanks for the pointer, I tried with `string` (without `[]`), but the openai python client gave the following error:

```
openai.BadRequestError: Error code: 400 - {'object': 'error', 'message': "1 validation error for EmbeddingResponseData\nembedding\n  Input should be a valid list [type=list_type, input_value=b'AAAAAADchD8AAAAAABCUPwA...wAAAAAA6JC/AAAAAAAIgD8=', input_type=bytes]\n    For further information visit https://errors.pydantic.dev/2.7/v/list_type", 'type': 'BadRequestError', 'param': None, 'code': 400}
``` 
*** Input should be a valid list*** (that was why I changed to array, but it further checks if the content in the array is `number`)

Reply:
I think we should update the definition of `EmbeddingResponseData` to allow both types of outputs